### PR TITLE
Category module task

### PR DESCRIPTION
**Pull request by MuzzammilJamil**
commit for category module CRUD.  but search missing in it as it will be pushed later in a while.
commit for generic API routing for whole project.
commit for secure routing for components.
commit for home navigation after success login.

### CODE REVIEW

File: src/Routes.js [hunk 1]
```diff
@@ -13,67 +13,59 @@ import TaxAdd from "./components/pages/taxes/taxAdd";
...
 const Routes = () => {
+
+  const renderWithLayout = (Component, props) => (
+    <AppShell {...props}>
+      <Component />
+    </AppShell>
+  );
+  
+  const authRenderWithLayout = (Component, props) => {
+    const readFromLocalStorage =  JSON.parse(window.localStorage.getItem('user'));
```
Comment (abdullahamin0111) [inline src/Routes.js:29]: use util function here to fetch user data from local storage

File: src/Routes.js [hunk 2]
```diff
@@ -13,67 +13,59 @@ import TaxAdd from "./components/pages/taxes/taxAdd";
...
+    const readFromLocalStorage =  JSON.parse(window.localStorage.getItem('user'));    
+    return <AppShell {...props}>
+              {readFromLocalStorage == null
+                ?  <Component />
+                : <Redirect to='/dashboard' />}
+            </AppShell>
+  }
+  
+  const PrivateRoute = ({ component: Component, ...rest }) => {
+    const readFromLocalStorage =  JSON.parse(window.localStorage.getItem('user'));
```
Comment (abdullahamin0111) [inline src/Routes.js:38]: use util function here to fetch user data from local storage

File: src/components/molecules/menu/index.js [hunk 3]
```diff
@@ -19,6 +20,7 @@ import {
 
 const SideMenu = () => {
   const history = useHistory();
+  const readFromLocalStorage = JSON.parse(getDataFromLocalStorage('user').data);
```
Comment (abdullahamin0111) [inline src/components/molecules/menu/index.js:23]: add a check if data is not found for user

File: src/components/pages/appShell/index.js [hunk 4]
```diff
@@ -18,9 +20,33 @@ const AppShell = (props) => {
...
   const { Header, Sider, Content } = Layout;
 
+  const history = useHistory();
+
   const toggle = () => {
     setCollapsed((s) => !s);
   };
+  
+  const toggleLogout= () => {
+    localStorage.clear();
```
Comment (abdullahamin0111) [inline src/components/pages/appShell/index.js:30]: use this in local-storage-util and call that function here

**Comment by MuzzammilJamil** [comment]
@abdullahamin0111 sir now you can review this PR as roues are updated with the new structure as per the recommended approach.

**Pull request merged**